### PR TITLE
[Pods] Adjust service actions to support pods

### PR DESCRIPTION
**Pull request by orlandohohmeier**
Adjust the service actions to support pods. I added mocked versions to enable proper testing.

_Please excuse the rather large number of changes. I figured it might be easier to land this with only one PR instead of a number smaller ones as I'm on vacation and wouldn't be able to do the rebase dance._

### CODE REVIEW

File: src/js/components/ServiceDetailConfigurationTab.js [hunk 1]
```diff
@@ -52,13 +53,16 @@ class ServiceDetailConfigurationTab extends React.Component {
   }
 
   handleApplyButtonClick() {
+    let {service} = this.props;
+
     let serviceConfiguration =
-      this.props.service.getVersions().get(this.state.selectedVersionID);
+        this.props.service.getVersions().get(this.state.selectedVersionID);
```
Comment (wavesoft) [inline src/js/components/ServiceDetailConfigurationTab.js]: `this.props.service` -> `service`

**Comment by wavesoft** [comment]
Besides a tiny #nit, looks good to me!
Ps. Love the code-to-test ratio 😄  👍

**Comment by wavesoft** [comment]
**mesosphere-ci** retest this please

File: src/js/components/ServiceDetailConfigurationTab.js (hunk 1 above)
Comment (orlandohohmeier) [inline src/js/components/ServiceDetailConfigurationTab.js]: ✔️ Adjusted

**Comment by mlunoe** [comment]
This needs a rebase. I couldn't rebase it, could you give it a try @orlandohohmeier?

File: src/js/events/MarathonActions.js [hunk 2]
```diff
@@ -133,9 +150,28 @@ var MarathonActions = {
...
+   * @param {Service} service - the service you want to delete
+   */
+  deleteService(service) {
+    if (!(service instanceof Service)) {
+      if (process.env.NODE_ENV !== 'production') {
+        throw new TypeError('service is not an instance of Service');
+      }
+
+      return;
+    }
```
Comment (philipnrmn) [inline src/js/events/MarathonActions.js]: I like the defensiveness here. I wonder whether we should adopt this as a pattern in general?

**Pull request merged**